app/client/src/components/shared/DataContent.js

```diff
@@ -113,515 +125,101 @@ function DataContent() {
     };
   }, []);
 
+  const { data, status } = useDataSourcesContext();
+
+  useLayoutEffect(() => {
```

Comment:
```suggestion
  useEffect(() => {
```

Is there a specific reason you used `useLayoutEffect` here? It seems to work fine with just a plain `useEffect`. 

The react documentation says that when possible to prefer using `useEffect` over `useLayoutEffect`. The docs also say that `useLayoutEffect` can negatively impact performance.

![image](https://user-images.githubusercontent.com/46329268/215822094-f2eb7fb5-5b48-42f4-8177-31b4a55974ae.png)

Reply:
Yeah I think I had it mixed up. I thought it fired after all elements are rendered (so all the `span`s would be in place), but it looks like it instead concerns layout measurements.